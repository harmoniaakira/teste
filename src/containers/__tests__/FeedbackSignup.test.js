import SignupFeedBack from "containers/Signup/Feedback";
import SignupAppraisal from "containers/Signup/Feedback/Appraisal";
import SignupError from "containers/Signup/Feedback/Error";
import SignupSuccess from "containers/Signup/Feedback/Success";
import SignupWarning from "containers/Signup/Feedback/Warning";
import { render, shallow } from 'enzyme';
import React from "react";


describe("FeedbackSignup", () => {
    const signupFeedBack = render(<SignupFeedBack shouldRender={true} />)

    it("should not render SignupFeedBack", () => {
        const feedbackNull = shallow(<SignupFeedBack shouldRender={false} />)
        expect(feedbackNull.get(0)).toBeFalsy()
    })

    it("should render SignupFeedBack", () => {
        expect(signupFeedBack.get(0)).toBeTruthy()
    })

    it("should match snaptshot", () => {
        expect(signupFeedBack).toMatchSnapshot()
    })
})

describe("FeedbackSignup Appraisal", () => {
    const signupAppraisal = shallow(<SignupAppraisal />)

    it("should match snaptshot", () => {
        expect(signupAppraisal).toMatchSnapshot()
    })
})

describe("FeedbackSignup Error", () => {
    const signupError = shallow(<SignupError />)

    it("should match snaptshot", () => {
        expect(signupError).toMatchSnapshot()
    })
})

describe("FeedbackSignup Warning", () => {
    const signupWarning = shallow(<SignupWarning />)

    it("should match snaptshot", () => {
        expect(signupWarning).toMatchSnapshot()
    })
})

describe("FeedbackSignup Success", () => {
    const signupSuccess = shallow(<SignupSuccess />)

    it("should match snaptshot", () => {
        expect(signupSuccess).toMatchSnapshot()
    })
})