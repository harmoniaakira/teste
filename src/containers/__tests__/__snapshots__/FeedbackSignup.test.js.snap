// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackSignup Appraisal should match snaptshot 1`] = `
<styled.div>
  <styled.div>
     Quase acabando! 
  </styled.div>
  <styled.div>
     Estamos conferindo os dados informados, leva só alguns instantes. ;) 
  </styled.div>
  <Loader />
</styled.div>
`;

exports[`FeedbackSignup Error should match snaptshot 1`] = `
<Feedback
  btnTxt="Alterar"
  icon="ERROR"
  onclick={[Function]}
  text="Confirme se a Data de Nascimento e o número de telefone estão corretos. Caso não esteja, realize a alteração."
  title="Ops! Confira os dados"
/>
`;

exports[`FeedbackSignup Success should match snaptshot 1`] = `
<Feedback
  btnTxt="Continuar"
  icon="SUCCESS"
  onclick={[Function]}
  text="Agora seu cliente receberá um token para validar o celular informado. ;) "
  title="Dados verificados com sucesso!"
/>
`;

exports[`FeedbackSignup Warning should match snaptshot 1`] = `
<Feedback
  btnTxt="Alterar"
  icon="WARNING"
  onclick={[Function]}
  text="Se o telefone utilizado pertence à loja/estabelecimento, alterar o telefone."
  title="Ops! Confira os dados"
/>
`;

exports[`FeedbackSignup should match snaptshot 1`] = `
Array [
  <div
    class="dtccwe"
  >
    <img
      class="bfFoYY"
      src="SvgrURL"
    />
  </div>,
  <div
    class="bNyUfp"
  >
    <div
      class="kEaUGk"
    >
      <div
        class="cquwcG"
      >
         Quase acabando! 
      </div>
      <div
        class="kGOuSw"
      >
         Estamos conferindo os dados informados, leva só alguns instantes. ;) 
      </div>
      <div
        class="dVqbqG"
      >
        <div />
      </div>
    </div>
    <div
      style="display:flex;gap:80px;margin:20px"
    >
      <button
        class="fNhGOU"
      >
         Sucesso 
      </button>
      <button
        class="fNhGOU"
      >
          Erro 
      </button>
      <button
        class="fNhGOU"
      >
          AVISO 
      </button>
    </div>
  </div>,
]
`;
